app/controllers/oidc_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# OidcController is used for performing an OpenID Connect auth
+# flow against Dex, allowing Velum to generate a complete kubeconfig
+# file for a user, including their JWT token
+class OidcController < ApplicationController
+  # allow anyone to start this, the auth will happen inside
+  # Dex.
+  skip_before_action :authenticate_user!  
+
+  def start
+    redirect_to authorization_uri(SecureRandom.hex(16))
+  end
+
+  def done
+    render "start.html.erb"
+  end
+
+  def authorization_uri(nonce)
+    client = client()
+    client.redirect_uri = url_for(controller: "oidc", action: "done")
+    client.authorization_uri(
+      response_type: :code,
+      nonce: nonce,
+      state: nonce,
+      scope: [:openid, :profile, :email, :offline_access, :groups].collect(&:to_s))    
+  end
+
+  def client
+    OpenIDConnect.http_config do |config|
+      config.ssl_config.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
Do we need to skip verification if we are adding the trusted CA now (with the changes below and on `bin/init`)? I understand this can be removed so it's actually verified.

Reply:
There's something wrong and the trusted CA doesn't work. Basically, the ruby httpclient library does not use the system trust store. It ships with its own. The line below this one that adds the system trust store isn't working. I need to dig in further.